python/ray/serve/_private/common.py

```diff
@@ -102,30 +105,53 @@ class DeploymentStatus(str, Enum):
     UPDATING = "UPDATING"
     HEALTHY = "HEALTHY"
     UNHEALTHY = "UNHEALTHY"
+    UPSCALING = "UPSCALING"
+    DOWNSCALING = "DOWNSCALING"
+
+
+class DeploymentStatusTrigger(str, Enum):
+    UNSPECIFIED = "UNSPECIFIED"
+    DEPLOY = "DEPLOY"
+    CONFIG_UPDATE = "CONFIG_UPDATE"
```

Comment:
what's the difference between these two?

Reply:
oh is "DEPLOY" a heavyweight config change?

Reply:
DEPLOY is when a new deployment is created, CONFIG_UPDATE is when an existing deployment is being updated with new values. (which is why when the target state has no target info yet, [I set the trigger to DEPLOY](https://github.com/ray-project/ray/pull/40951/files#diff-c4c2583a4c2f3a3c87ada6faebd0b2dfef404e165df133a11195be5d65fcb387R1465-R1470).

Reply:
I see